crates/router/src/connector/paypal/transformers.rs

```diff
@@ -486,7 +486,12 @@ impl TryFrom<&PaypalRouterData<&types::PaymentsAuthorizeRouterData>> for PaypalP
                             experience_context: ContextStruct {
                                 return_url: item.router_data.request.complete_authorize_url.clone(),
                                 cancel_url: item.router_data.request.complete_authorize_url.clone(),
-                                shipping_preference: ShippingPreference::SetProvidedAddress,
+                                shipping_preference: if item.router_data.address.shipping.is_some()
+                                {
+                                    ShippingPreference::SetProvidedAddress
+                                } else {
+                                    ShippingPreference::GetFromFile
```

Comment:
What happens if we GetFromFile?

Reply:
when we skip this shipping_preference -- then default value is considered as GET_FROM_FILE

Reply:
-  When it comes to SHIPPING Preference -- if you have shipping address and passing in the request paylaod you can always pass it as SET_PROVIDED_ADDRESS
```
"experience_context": {
                "user_action": "PAY_NOW",
                "shipping_preference": "SET_PROVIDED_ADDRESS",
                "return_url": "https://www.website-url.com/return",
                "cancel_url": "https://www.website-url.com/cancel"}
```
- When it comes to SHIPPING Preference -- If you are sure if it DIGITAL then only pass NO_SHIPPING
```
"experience_context": {
                "user_action": "NO_SHIPPING",
                "shipping_preference": "SET_PROVIDED_ADDRESS",
                "return_url": "https://www.website-url.com/return",
                "cancel_url": "https://www.website-url.com/cancel"}
```
-  And if you are unsure if its DIGITAL and no shipping address provided you can pass default value `GET_FROM_FILE`
```
"experience_context": {
                "user_action": "PAY_NOW",
                "shipping_preference": "GET_FROM_FILE",
                "return_url": "https://www.website-url.com/return",
                "cancel_url": "https://www.website-url.com/cancel"}
```
Or can skip this shipping_preference -- because default value is considered as `GET_FROM_FILE`